src/test/groovy/graphql/nadel/NadelE2ETest.groovy

```diff
@@ -984,4 +986,66 @@ class NadelE2ETest extends Specification {
         ResultNodesUtil.toExecutionResult(originalExecutionResult).errors.isEmpty()
     }
 
+    def "object type from underlying schema is removed"() {
+
+        def simpleNDSL = '''
+         service MyService {
+            type Query {
+                hello: World
+            }
+            
+            interface World {
+                id: ID
+                name: String
+            }
+            
+            type Mars implements World {
+                id: ID
+                name: String
+            }
+         }
+        '''
+
+        def underlyingSchemaChanged = typeDefinitions('''
+            type Query{
+                hello: World  
+            } 
+            
+            interface World {
+                id: ID
+                name: String
+            }
+        ''')
+
+        def query = '''
+        query OpName { 
+            hello {
+                ... on Mars {
+                    name
+                }
+            } 
+        }
+        '''
+
+        def nadel = newNadel()
+                .dsl(simpleNDSL)
+                .serviceExecutionFactory(TestUtil.serviceFactory(delegatedExecution, underlyingSchemaChanged))
+                .build()
+
+        def nadelExecutionInput = newNadelExecutionInput()
+                .query(query)
+                .operationName("OpName")
+                .build()
+
+        when:
+        nadel.execute(nadelExecutionInput).join()
+        then:
+        def e = thrown CompletionException
+        e.cause instanceof AssertException
+        e.cause.message == "Schema mismatch: Underlying schema is missing required field Mars"
```

Comment:
this error message doesn't seem completely right: `Mars` is the type not the field.
Maybe we need to differentiate between type and field missing.

Reply:
Yeah sorry this should be Object/Interface